File: KYForm/KYForm/Demos/ViewController.m
//
//  ViewController.m
//  KYForm
//
//  Created by mac on 2017/12/18.
//

#import "ViewController.h"
#import "PersonalViewController.h"
#import "SetttingViewController.h"
#import "LoanViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

@property (nonatomic, strong) UITableView    *tableView;
@property (nonatomic, strong) NSMutableArray *dataSource;

@end

@implementation ViewController

- (instancetype)init {
    if (self = [super init]) {
    }
    
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Demos";
    self.view.backgroundColor = [UIColor whiteColor];
    
    [self.view addSubview:self.tableView];
    [self.tableView reloadData];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *identifier = @"demoCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:identifier];
    }
    
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    cell.textLabel.text = [self.dataSource objectAtIndex:indexPath.row];
    
    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    NSInteger row = indexPath.row;
    if (row == 0) {
        [self.navigationController pushViewController:[PersonalViewController new] animated:YES];
    } else if (row == 1) {
        [self.navigationController pushViewController:[SetttingViewController new] animated:YES];
    } else {
        [self.navigationController pushViewController:[LoanViewController new] animated:YES];
    }
}

- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
        _tableView.backgroundColor = [UIColor whiteColor];
        _tableView.tableFooterView = [UIView new];
        _tableView.dataSource = self;
        _tableView.delegate = self;
    }
    
    return _tableView;
}

- (NSMutableArray *)dataSource {
    if (!_dataSource) {
        _dataSource = [NSMutableArray arrayWithArray:@[@"个人中心", @"设置中心", @"贷款机构"]];
    }
    
    return _dataSource;
}

@end

File: Charting/Charting/ViewController.swift
//
//  ViewController.swift
//  Charting
//
//  Created by kaideyi on 2017/3/5.
//  Copyright © 2017年 kaideyi.com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    public lazy var tableView: UITableView = {
        let tb = UITableView()
        tb.frame = CGRect(x: 0, y: 0, width: self.view.width, height: self.view.height)
        tb.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "cell")
        tb.tableFooterView = UIView()
        tb.dataSource = self
        tb.delegate = self
        
        self.view.addSubview(tb)
        
        return tb
    }()
    
    var data: NSArray = ["1.柱形图", "2.折线图", "3.饼状图"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Charting"
        tableView.reloadData()
    }
}

// MARK: -
extension ViewController: UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        cell.textLabel?.text = data[indexPath.row] as? String
        return cell
    }
}

extension ViewController: UITableViewDelegate {
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        let controller = SubViewController(indexPath.row)
        self.navigationController?.pushViewController(controller, animated: true)
    }
}

